bundle/edu.gemini.itc/src/main/scala/edu/gemini/itc/service/ItcServiceImpl.scala

```diff
@@ -28,25 +27,33 @@ class ItcServiceImpl extends ItcService {
 
   def calculate(p: ItcParameters): Result = try {
 
-    // Get the SED data from an aux file. For now we can assume that the ITC service is running on the same
-    // machine as the database (localhost). In case this setup changes, we need to change this here, too.
-    def auxFileDistribution(id: String, name: String): SpectralDistribution = {
-      val programId     = SPProgramID.toProgramID(id)
-      val spectrumBytes = new AuxFileClient("localhost", 8443).fetchToMemory(programId, name)
-      val spectrum      = new String(spectrumBytes)
-      UserDefinedSpectrum(name, spectrum)
+    // update parameters sent from client with stuff that needs to be done on the server
+    val updatedParams: ItcParameters = {
+
+      // Get the SED data from an aux file. For now we can assume that the ITC service is running on the same
+      // machine as the database (localhost). In case this setup changes, we need to change this here, too.
+      def readAuxFile(id: String, name: String): SpectralDistribution = {
+        val programId     = SPProgramID.toProgramID(id)
+        val spectrumBytes = new AuxFileClient("localhost", 8443).fetchToMemory(programId, name)
```

Comment:
You mean localhost and port number? Yeah, that's a point. But making it configurable only helps if people remember to change the configuration.. pick your poison, I guess?